### ISSUE TITLE
Test artifacts configured using config-repo is uploaded to GoCD server's build artifacts' default location.

### ISSUE THREAD

**Issue by bdpiparva** [gocd/gocd#4071]
##### Issue Type
- Bug Report

##### Summary

I have job with artifacts configured using json config repo plugin(https://github.com/tomzo/gocd-json-config-plugin). Here is the configuration

```json
{
  "environment_variables": [],
  "format_version": 1,
  "group": "configrepo-example",
  "materials": [
    {
      "type": "git",
      "url": "https://github.com/bdpiparva/config_repo_sample"
    }
  ],
  "name": "firstpipe",
  "stages": [
    {
      "clean_working_directory": false,
      "environment_variables": [],
      "fetch_materials": true,
      "jobs": [
        {
          "artifacts": [
            {
              "source": "*.json",
              "type": "test"
            },
            {
              "source": "*.zip",
              "type": "build"
            }
          ],
          "environment_variables": [],
          "name": "build",
          "properties": [],
          "resources": [],
          "run_instance_count": null,
          "tabs": [],
          "tasks": [
            {
              "arguments": [],
              "command": "ls",
              "on_cancel": null,
              "run_if": "passed",
              "type": "exec",
              "working_directory": null
            }
          ],
          "timeout": 0
        }
      ],
      "name": "build",
      "never_cleanup_artifacts": false
    }
  ]
}
```

By default it should upload test artifacts under `testoutput` folder but this is not true in case of if pipeline is coming from config-repo.

@tomzo --- is this intentional behaviour ?. I can see(https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/config/ConfigConverter.java#L501-L504) that all artifact without destination directory is uploaded in the artifacts directory instead of `testoutput` dir.

**Comment by tomzo** [comment]
Hi,

> By default it should upload test artifacts under testoutput folder but this is not true in case of if pipeline is coming from config-repo.

I did not know about this behavior until now. So it is a bug. As you noticed that should be fixed in `ConfigConverter` somewhere around those lines.

**Comment by bdpiparva** [comment]
This issue is fixed in #4083
